### PR TITLE
Fix resourceCache to return proper modules when PS modules are installed during run

### PR DESCRIPTION
**Pull request by Gijsreyn**
# PR Summary

This PR addresses the issue described in #745. When modules are installed during a document being applied, the subsequent modules that are checked in the cache are overwritten. Example:

1. The first run initiates the resource cache (dsc resource list) as no cache is found.
2. The first configuration is applied, installing new PowerShell modules (triggering the cache on module paths to be stale).
3. The assertion triggers the `Invoke-DscCacheRefresh -Module '<module>'`, flushing out the cache with only that particular module.
4. Modules underneath aren't found anymore in the same sequence.

It's a pretty rough edge case and quite interesting to debug. Nevertheless, there isn't any test added because it would either introduce an external dependency or require internal work to be implemented. Of course, I can be wrong on this and would appreciate some help to write one.

### CODE REVIEW

**Comment by Gijsreyn** [comment]
@SteveL-MSFT, if you have some time, you might taking a glimpse?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
Can you repro the original issue by having a test case that executes a test resource that adds a new test resource?

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
-            "name": "Debug sshdconfig",
-            "type": "cppvsdbg",
-            "request": "launch",
-            "program": "${workspaceFolder}/sshdconfig/target/debug/sshdconfig.exe",
-            "args": ["get"],
-            "cwd": "${workspaceFolder}"
-        }
-    ]
-}
+// {
```
Comment (SteveL-MSFT) [inline .vscode/launch.json:1]: Why was this included and commented out?

File: powershell-adapter/Tests/win_powershellgroup.tests.ps1 [hunk 2]
```diff
@@ -25,67 +25,154 @@ Describe 'WindowsPowerShell adapter resource tests - requires elevated permissio
...
-
-        $r = dsc resource list --adapter Microsoft.Windows/WindowsPowerShell
-        $LASTEXITCODE | Should -Be 0
-        $resources = $r | ConvertFrom-Json
-        ($resources | ? {$_.Type -eq 'PSDesiredStateConfiguration/File'}).Count | Should -Be 1
-    }
-
-    It 'Get works on Binary "File" resource' -Skip:(!$IsWindows){
-
+    # It 'Windows PowerShell adapter supports File resource' -Skip:(!$IsWindows){
```
Comment (SteveL-MSFT) [inline powershell-adapter/Tests/win_powershellgroup.tests.ps1:28]: Why are these tests commented out?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: powershell-adapter/Tests/win_powershellgroup.tests.ps1 [hunk 3]
```diff
@@ -82,12 +81,100 @@ Describe 'WindowsPowerShell adapter resource tests - requires elevated permissio
...
+    properties: 
+      Name: NuGet
+    dependsOn:
+      - "[resourceId('Microsoft.Windows/WindowsPowerShell', 'File')]"
+      - "[resourceId('Microsoft.DSC/Assertion', 'File present')]"
+"@
+        # output to file for Windows PowerShell 5.1
+        $filePath = "$testdrive\test.assertion.dsc.resource.yaml"
+        $yaml | Set-Content -Path $filePath -Force
+        dsc config test -f $filePath 2> "$TestDrive/error.txt"
```
Comment (SteveL-MSFT) [inline powershell-adapter/Tests/win_powershellgroup.tests.ps1:170]: No change needed but more for your info, you can always pass in `$yaml` to `--input` instead of writing it as a file:

```powershell
dsc config test -i $yaml
```

**Review by SteveL-MSFT — APPROVED** [review]

File: powershell-adapter/Tests/win_powershellgroup.tests.ps1 (hunk 3 above)
Comment (Gijsreyn) [reply powershell-adapter/Tests/win_powershellgroup.tests.ps1:170]: I expected it also, yet, there was an issue: #752.
